cmd/identityoracled/handlers/identity.go

```diff
@@ -90,11 +99,18 @@ func (v *Verify) XPubSignature(ctx context.Context, log logger.Logger, w http.Re
 		return translate(errors.Wrap(err, "unmarshal request"))
 	}
 
+	for _, xpub := range requestData.XPubs {
+		if xpub.IsPrivate() {
+			web.Respond(ctx, log, w, "private keys not allowed", http.StatusUnprocessableEntity)
+			return nil
+		}
+	}
+
 	dbConn := v.MasterDB.Copy()
 	defer dbConn.Close()
 
 	// Verify that the public key is associated with the entity.
-	sigHash, height, approved, err := oracle.VerifyXPub(ctx, dbConn, v.BlockHandler,
+	sigHash, height, approved, description, err := oracle.VerifyXPub(ctx, dbConn, v.BlockHandler,
```

Comment:
That's a lot of return elements.

How about a struct and an error ?

Reply:
It just kind of evolved like that. I will fix it.